api/bots_api/bots_test_lib.py

```diff
@@ -72,22 +73,35 @@ def get_bot_message_handler(self):
         message_handler = self.bot_to_run(bot_module)
         return message_handler
 
-    def assert_bot_output(self, messages, bot_response, expected_method):
-        # type: (List[Dict[str, Any]], List[Dict[str, str]], str) -> None
+    def assert_bot_output(self, messages, bot_response, expected_method,
+                          http_request=None, http_response=None):
+        # type: (List[Dict[str, Any]], List[Dict[str, str]], str, Optional[Dict[str, Any]], Optional[Dict[str, Any]]) -> None
         message_handler = self.get_bot_message_handler()
         # Mocking BotHandlerApi
         with patch('bots_api.bot_lib.BotHandlerApi') as MockClass:
             instance = MockClass.return_value
 
-            for (message, response) in zip(messages, bot_response):
-                # Send message to the concerned bot
-                message_handler.handle_message(message, MockClass(), StateHandler())
-                # Check if the bot is sending a message via `send_message` function.
-                # Where response is a dictionary here.
-                if expected_method == "send_message":
-                    instance.send_message.assert_called_with(response)
-                else:
-                    instance.send_reply.assert_called_with(message, response['content'])
+            with patch('requests.get') as mock_get:
+                mock_result = mock.MagicMock()
+                mock_result.json.return_value = http_response
```

Comment:
Do I understand it correctly that by setting `return_value` to `http_response`, every message-response query will have the same mock http_response? If so, I think there should be a distinct mock http_response for every query, that can e.g. be passed in the expectations dict.